modules/RTC/RTCUtils.js

```diff
@@ -1147,20 +1147,19 @@ class RTCUtils extends Listenable {
                         + 'not supported on this browser.');
                 }
 
-                return new Promise((resolve, reject) => {
-                    screenObtainer.obtainStream(
-                        this._parseDesktopSharingOptions(options),
-                        desktop => resolve({
-                            audioVideo,
-                            desktop
-                        }),
-                        error => {
-                            if (audioVideo) {
-                                this.stopMediaStream(audioVideo);
-                            }
-                            reject(error);
-                        });
-                });
+                return this._newGetDesktopMedia(this._parseDesktopSharingOptions(options))
```

Comment:
Let's avoid adding the `_new` functions to the "old" ones. I see that `_newGetDesktopMedia` is functionally equivalent to what we were doing before. Why the change?